src/test/java/com/jfrog/ide/idea/scan/YarnScannerTest.java

```diff
@@ -0,0 +1,113 @@
+package com.jfrog.ide.idea.scan;
+
+import com.intellij.testFramework.HeavyPlatformTestCase;
+import com.intellij.util.ConcurrencyUtil;
+import com.jfrog.ide.common.deptree.DepTree;
+import com.jfrog.ide.common.nodes.DependencyNode;
+import com.jfrog.ide.common.nodes.FileTreeNode;
+import com.jfrog.ide.common.scan.GraphScanLogic;
+import org.apache.commons.io.FileUtils;
+import org.jetbrains.plugins.gradle.service.project.open.GradleProjectImportUtil;
+import org.jfrog.build.api.util.NullLog;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+
+public class YarnScannerTest extends HeavyPlatformTestCase {
+    private static final Path YARN_ROOT = Paths.get(".").toAbsolutePath().normalize().resolve(Paths.get("src", "test", "resources", "yarn"));
+    private ExecutorService executorService;
+    private String tempProjectDir;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        tempProjectDir = createTempProjectDir("exampleYarnPackage");
+        executorService = ConcurrencyUtil.newSameThreadExecutorService();
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        try {
+            executorService.shutdown();
+        } finally {
+            // Ensure that tearDown gets executed even if an exception is thrown
+            super.tearDown();
+        }
+    }
+
+    private String createTempProjectDir(String projectName) throws IOException {
```

Comment:
projectName is always "exampleYarnPackage".
* Let's ensure to eliminate all compiler warnings from the code.
* Put the "exampleYarnPackage" in a private static class variable